overview/authentication-flows.mdx

```diff
@@ -0,0 +1,195 @@
+
+    ### Security Benefits
+    - ✅ **Maximum security** through dual authentication
+    - ✅ Protection against authorization code interception
+    - ✅ Client authentication via secret
+    - ✅ Dynamic PKCE verification per request
+
+    ### Configuration
+    ```javascript
+    const config = {
+      clientId: "your_client_id",
+      clientSecret: "your_client_secret", // Contact Civic for this
+      pkce: true, // Enable PKCE alongside client secret
+      redirectUrl: "https://yourapp.com/callback"
+    }
+    ```
+  </Tab>
+</Tabs>
+
+## When to Use Each Method
+
+<CardGroup cols={3}>
+  <Card title="PKCE Only" icon="browser">
+    - Single-page applications (SPAs)
+    - Mobile applications
+    - Frontend applications (React, Vue, Angular)
+    - When client secrets cannot be securely stored
+    - Default for public clients
+  </Card>
+  <Card title="Client Secret Only" icon="server">
+    - Legacy OAuth 2.0 integrations
+    - Traditional server-side applications
+    - When PKCE is not supported
+    - Simple backend services
+    - Existing OAuth infrastructure
+  </Card>
+  <Card title="PKCE + Client Secret" icon="shield">
+    - **High-security server applications**
+    - Financial or healthcare systems
+    - Enterprise backend services
+    - **Maximum security requirements**
+    - Modern confidential clients
+  </Card>
+</CardGroup>
+
+## Getting Client Secrets
+
+<Info>
+If you need a client secret, contact [support@civic.com](mailto:support@civic.com).
+</Info>
+
+<Warning>
+**Security Requirements**: Client secrets must be stored securely and never exposed in client-side code. They are suitable only for server-side applications with secure credential storage.
+</Warning>
+
+<Info>
+**Flexible Security**: Choose the authentication method that best fits your application architecture. Use PKCE-only for public clients, client secrets for traditional OAuth compatibility, or both together for maximum security.
+</Info>
+
+## Security Considerations
+
+<AccordionGroup>
+  <Accordion title="PKCE Security Model">
+    - **No secrets to compromise:** Even if your frontend code is inspected, there are no sensitive credentials
+    - **Dynamic verification:** Each authentication request uses a unique code verifier/challenge pair
+    - **Domain validation:** Production apps require domain registration for additional security
+    - **Industry standard:** Recommended by OAuth 2.0 Security Best Practices
+  </Accordion>
+
+  <Accordion title="Client Secret Security Model">
+    - **Secure storage required:** Client secrets must be stored securely and never exposed to client-side code
+    - **Server-side only:** Authentication logic must run on your backend servers
+    - **Environment variables:** Use environment variables or secure secret management systems
+    - **HTTPS required:** Always use HTTPS in production to protect credentials in transit
+  </Accordion>
+</AccordionGroup>
+
+## Implementation Examples
```

Comment:
rather than duplicating code (imports, constructor etc) can we just have a table of config parameters?